source/base/utilities.cc

```diff
@@ -123,6 +124,37 @@ namespace Utilities
     return std::string( input, start_idx, end_idx+1-start_idx);
   }
 
+
+
+  std::string
+  trim_whitespace(const std::string &input)
+  {
+    std::string::size_type left = 0;
+    std::string::size_type right = input.size() - 1;
+
+    std::locale current_locale;
+
+    for (; left < input.size(); ++left)
+      {
+        if (!std::isspace(input[left], current_locale))
```

Comment:
http://en.cppreference.com/w/cpp/string/byte/isspace says "classified by the currently installed C locale" which looks like is the same thing you are doing or not? Looking at the table at the bottom of the page it looks like this is exactly what you want...